src/titiler/multidim/factory.py

```diff
@@ -0,0 +1,181 @@
+"""TiTiler.xarray factory."""
+
+from typing import List, Literal, Optional, Type
+from urllib.parse import urlencode
+
+import jinja2
+import numpy as np
+from attrs import define
+from fastapi import Depends, Query
+from starlette.requests import Request
+from starlette.responses import HTMLResponse
+from starlette.templating import Jinja2Templates
+from typing_extensions import Annotated
+
+from titiler.core.dependencies import ColorFormulaParams, DefaultDependency
+from titiler.core.resources.enums import ImageType
+from titiler.core.resources.responses import JSONResponse
+from titiler.multidim.reader import XarrayReader
```

Comment:
non-blocking but now that I'm looking at this I am wondering if the naming convention should help distinguish local application dependencies vs imported dependencies... like titiler_app.multidim|xarray or titiler.app.multidim|xarrat